index.rst

```diff
@@ -1,61 +1,311 @@
 
-   Feel free to delete this instructional comment.
+Data access changes
+===================
 
-:tocdepth: 1
+The consolidation of data access through butler gen3 represents an important change in the interface and operation of the MTAOS.
+The following is a series of improvements that can be implemented to improve the MTAOS usability.
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+Processing calibration products
+-------------------------------
 
-.. sectnum::
+This is currently done via the ``processCalibrationProducts`` command, which expected a directory containing the raw calibration data.
+Instead, the butler instances hosting the data will also contain processed calibration products suitable for all exposures.
+This will have to be carefully managed by observatory personnel over time, but is a much more suitable approach than leaving the MTAOS in charge of the process.
 
-.. TODO: Delete the note below before merging new content to the master branch.
+Image ingestion
+---------------
 
-.. note::
+As part of the wavefront processing commands, the current version of the MTAOS receives files names and directory location.
+This data is initially ingested to a butler instance to allow them to be ISR corrected and later WEP processed.
+
+Since the data will now be ingested by a different component and will be made available to the MTAOS through butler, there is no more need for a local ingestion.
+
+Simplification of wfs process commands
+--------------------------------------
+
+As mention above, the wfs process commands receives file names and path.
+With the data already ingested in the butler the images can be uniquely identified by the ``visitId`` entry in the database.
+That means, all process commands need to be updated to receive the ``visitId`` instead of the files name and path.
+
+At the same time, most of the metadata that is now passed as part of the process commands (``fieldRA``, ``fieldDEC``, ``filter``, ``cameraRotation``) is also available as metadata in the butler and can be removed from the command payload.
+
+Furthermore, through the ``visitId`` one can identify the source of the image (ComCam or LSSTCam), so there is no need for special commands for each one of them.
+
+At the same time, we can differentiate between an Intra/Extra wavefront analysis process and a corner wavefront sensors by the input alone.
+If only one input ``visitId`` then it can be assumed to be a corner wavefront sensor run.
+If, in addition to ``visitId`` an ``extraId`` is given, it can be considered and Intra/Extra wavefront process.
+
+The command would be used as follows:
+
+.. code-block:: py
+
+  from lsst.ts import salobj
+
+  d = salobj.Domain()
+
+  mtaos = salobj.Remote(d, "MTAOS")
+
+  await mtaos.start_task
+
+  # Process full array intra/extra focal images.
+  # Frames can be ComCam or LSSTCam
+  await mtaos.cmd_runWEP.set_start(
+      visitId=2020103000040, extraId=2020103000041
+  )
```

Comment:
As explained above, the information here might not be enough.

Reply:
This is a good point. If we want to be able to run wep in more than one intra/extra image at a time we will need to provide a vector of `dataId`s.

Here I am only providing a quick snipped of code, assuming I already took the images and want to analyze them. But I see your point about source selection, etc...